calculator_test.go

```diff
@@ -2,23 +2,275 @@ package calculator_test
 
 import (
 	"calculator"
+	"math/rand"
 	"testing"
 )
 
-func TestAdd(t *testing.T) {
-	t.Parallel()
-	var want float64 = 4
-	got := calculator.Add(2, 2)
+func assertNumbers(want, got float64, t *testing.T) {
 	if want != got {
-		t.Errorf("want %f, got %f", want, got)
+		t.Fatalf("want %f, got %f", want, got)
+	}
+}
+
+func assertNumbersErrHandling(want, got float64, errExpected bool, t *testing.T) {
+	if (want != got) && !errExpected {
+		t.Fatalf("want %f, got %f", want, got)
+	}
+}
+
+func TestAdd(t *testing.T) {
+	var totalRandom int = 100
+
+	t.Run("Fixed number addition", func(t *testing.T) {
```

Comment:
I would break this up into two tests. First, test a slice of comprehensive test cases with fixed inputs, as for your other functions. Second, a `TestAddRandom` function which only tests random inputs.